### PR TITLE
add --ignore option to data_ingestion.py to ignore directories like .git

### PR DESCRIPTION
**Pull request by jondwillis**
### Background

I used data_ingestion.py to pre-seed my database, but it got bogged down ingesting dependencies and git files.
### Changes

Add --ignore to data_ingestion.py and update documentation
### Documentation

documented in the local README.md and in code comments
### Test Plan

N/A (see below)
### PR Quality Checklist
- [x] My pull request is atomic and focuses on a single change.
- [x] I have thoroughly tested my changes with multiple different prompts.
- [x] I have considered potential risks and mitigations for my changes.
- [x] I have documented my changes clearly and comprehensively.
- [x] I have not snuck in any "extra" small tweaks changes 

There are no existing tests that I found for data_ingestion.py

### CODE REVIEW

**Review by ravefaces — APPROVED** [review]

**Comment by jondwillis** [comment]
Rebased, force pushed. 

I also fixed an issue where the ignore list wasn't being sent with the search_files command.

**Review by hdkiller — APPROVED** [review]

File: autogpt/commands/file_operations.py [hunk 1]
```diff
@@ -210,12 +210,13 @@ def delete_file(filename: str) -> str:
         return f"Error: {str(e)}"
 
 
-@command("search_files", "Search Files", '"directory": "<directory>"')
-def search_files(directory: str) -> list[str]:
+@command("search_files", "Search Files", '"directory": "<directory>", "ignore_list": "<list_of_directories_to_ignore>')
```
Comment (Pwuts) [inline autogpt/commands/file_operations.py:213]: What kind of list? Comma separated? Space separated?

File: README.md [hunk 2]
```diff
@@ -412,6 +412,7 @@ options:
   -h, --help               show this help message and exit
   --file FILE              The file to ingest.
   --dir DIR                The directory containing the files to ingest.
+  --ignore                 For use in combination with --dir, a list of directories to ignore
```
Comment (Pwuts) [inline README.md:415]: Please include list format in instructions

File: data_ingestion.py [hunk 3]
```diff
@@ -46,6 +46,13 @@ def main() -> None:
     group.add_argument(
         "--dir", type=str, help="The directory containing the files to ingest."
     )
+    parser.add_argument(
+        "--ignore",
+        nargs="*",
+        type=str,
+        help="List of directories to ignore in combination with --dir, e.g. --ignore .git node_modules",
+        default=[],
+    )
```
Comment (Pwuts) [inline data_ingestion.py:55]: `--ignore foo --ignore bar`
or
`--ignore foo bar`
?

File: autogpt/commands/file_operations.py [hunk 4]
```diff
@@ -227,7 +228,10 @@ def search_files(directory: str) -> list[str]:
     else:
         search_directory = path_in_workspace(directory)
 
-    for root, _, files in os.walk(search_directory):
+    for root, dirs, files in os.walk(search_directory):
+        # Remove ignored directories from traversal
+        dirs[:] = [d for d in dirs if d not in ignore_list]
```
Comment (Pwuts) [inline autogpt/commands/file_operations.py:233]:
```suggestion
        dirs = [d for d in dirs if d not in ignore_list]
```

**Review by Pwuts — CHANGES_REQUESTED** [review]

**Comment by p-i-** [comment]
This is a mass message from the AutoGPT core team.
Our apologies for the ongoing delay in processing PRs.
This is because we are re-architecting the AutoGPT core!

For more details (and for infor on joining our Discord), please refer to:
  https://github.com/Significant-Gravitas/Auto-GPT/wiki/Architecting

**Comment by jondwillis** [comment]
@Pwuts you can go ahead and close this

**Pull request closed without merging**